flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/PhysicalTableSourceScan.scala

```diff
@@ -58,7 +58,13 @@ abstract class PhysicalTableSourceScan(
   }
 
   override def explainTerms(pw: RelWriter): RelWriter = {
-    super.explainTerms(pw).item("fields", getRowType.getFieldNames.asScala.mkString(", "))
+    val hasWatermark = tableSourceTable.watermarkSpec.isDefined && tableSourceTable.isStreamingMode
+    val rowtime = tableSourceTable.watermarkSpec.map(_.getRowtimeAttribute).orNull
```

Comment:
Does this mean batch mode may also have watermark definitions ?

Reply:
+1, we can define the watermark in batch scan, but it's useless and we don't have to show it in digest

Reply:
Yes. A source DDL can also be used in batch mode and streaming mode. But watermark is not needed in digest.